pyxem/signals/indexation_results.py

```diff
@@ -147,6 +159,487 @@ def _get_second_best_phase(z):
+            return self.map(orientation2phase, sizes=sizes, inplace=False).data
+        else:
+            return None
+
+    def to_single_phase_orientations(self, **kwargs) -> Orientation:
+        """Convert the orientation map to an `Orientation`-object,
+        given a single-phase simulation.
+        """
+        if self.simulation.has_multiple_phases:
+            raise ValueError(
+                "Multiple phases found in simulation (use to_crystal_map instead)"
+            )
+
+        # Use the quaternion data from rotations to support 2D rotations,
+        # i.e. unique rotations for each navigation position
+        rotations = hs.signals.Signal2D(self.simulation.rotations.data)
+
+        return Orientation(
+            self.map(
+                rotation_from_orientation_map,
+                rots=rotations,
+                inplace=False,
+                output_signal_size=(self.num_rows, 4),
+                output_dtype=float,
+                **kwargs,
+            ),
+            symmetry=self.simulation.phases.point_group,
+        )
+
+    def to_single_phase_vectors(
+        self, n_best_index: int = 0, **kwargs
+    ) -> hs.signals.Signal1D:
+        """
+        Get the reciprocal lattice vectors for a single-phase simulation.
+
+        Parameters
+        ----------
+        n_best_index: int
+            The index into the `n_best` matchese
+        """
+
+        if self.simulation.has_multiple_phases:
+            raise ValueError("Multiple phases found in simulation")
+
+        # Use vector data as signal in case of different vectors per navigation position
+        vectors_signal = hs.signals.Signal1D(self.simulation.coordinates)
+
+        return self.map(
+            extract_vectors_from_orientation_map,
+            all_vectors=vectors_signal,
+            inplace=False,
+            output_signal_size=(),
+            output_dtype=object,
+            show_progressbar=False,
+            **kwargs,
+        )
+
+    def to_crystal_map(self) -> CrystalMap:
+        """Convert the orientation map to an `orix.CrystalMap` object"""
+        if self.axes_manager.navigation_dimension != 2:
+            raise ValueError(
+                "Only 2D navigation supported. Please raise an issue if you are interested in "
+                "support for 3+ navigation dimensions."
+            )
+
+        x, y = [ax.axis for ax in self.axes_manager.navigation_axes]
+        xx, yy = np.meshgrid(x, y)
+        xx = xx - np.min(xx)
+        yy = yy - np.min(yy)
+        scan_unit = self.axes_manager.navigation_axes[0].units
+        rotations = self.to_rotation(flatten=True)
+        phase_index = self.to_phase_index()
+
+        if self.simulation.has_multiple_phases:
+            phases = PhaseList(list(self.simulation.phases))
+            if phase_index.ndim == 3:
+                phase_index = phase_index[..., 0]
+            phase_index = phase_index.flatten()
```

Comment:
@pc494 or @hakonanes We only pass the phase of the top matching pattern but I wonder what changes in orix would be necessary to have multiple phase indexs similar to how multiple rotations are allowed.

Reply:
I allowed multiple rotations because EMsoft, and eventually kikuchipy, returns those. I seldom use them myself, other than to generate a so-called [orientation similarity map](https://kikuchipy.org/en/stable/reference/generated/kikuchipy.indexing.orientation_similarity_map.html) (see the *Notes*).

Is it enough to return multiple crystal maps if a user wants to have a look at the lesser matching solutions? My fear is that this would be a little-used feature that adds a high degree of complexity to the crystal map implementation... 

Looking at workflows more broadly, the end-goal is a crystal map with the best possible solution for each point. My feeling is that evaluating the lesser matches should be done by something else than the end-crystal map.

Reply:
@hakonanes that sounds reasonable. I don't want to add too much complexity when you are right. Multiple crystal maps would solve the problem. 

I guess we need a couple more tools for refining the fit, although I'm not entirely sure what those are :)